pipelines/controller/src/main/java/com/google/fhir/analytics/HiveTableManager.java

```diff
@@ -76,7 +81,7 @@ public void createResourceAndCanonicalTables(
    * thriftServerParquetPath is the exact path for parquet files and resource shall be the
    * respective resource name e.g. Patient
    */
-  private void createTablesForResource(
+  private synchronized void createTablesForResource(
```

Comment:
I am hoping the table creations are faster, does the latency have any impact on the size of the parquet files because the lock will be hold for that long. Assuming this is negligible for now.

Reply:
I don't think that the latency of this function is a big concern but I have not tested this with a huge number of tables (maybe ~10 to 20). In any case, this is to fix a potential race condition; please see the TODO before `PipelineManager.PipelineThread.createHiveResourceTables` which is removed in this PR.